### ISSUE TITLE
[Backtest] Build Backtest Utility for Single Model

### ISSUE THREAD

**Issue by PricePredictor** [PricePredictor/alphapulse-backend#10]
**📝 Description:**  
Create a reusable backtest function to evaluate historical performance of any model on a given ticker using past data.

**📦 Deliverables:**  
- [ ] `services/backtest.py` module
- [ ] Function to simulate prediction vs actual
- [ ] Return RMSE, MAPE, and directional accuracy

**✅ Acceptance Criteria (Tests):**

| Test Scenario | How to Test | Expected Result |
|---------------|-------------|-----------------|
| Valid ticker | Run `backtest("TSLA", "xgb", days=30)` | Returns 3 metrics |
| Invalid ticker | Use `FAKE1` | Returns safe error or fallback |
| Model mismatch | Bad model name | Raises error |
| Metric check | Check MAPE < 100 | Must be reasonable |

**🛠 Tools Required:**  
`sklearn.metrics`, `pandas`, model files

**⏳ Estimate:**  
2.5 hours

**🔗 Milestone:**  
`Phase 1 – Model Evaluation`  
**🏷️ Label:** `analytics`

**Comment by PricePredictor** [comment]
# ✅ Ticket 10 – Closure Documentation

## 📝 Description
Create a reusable backtest function to evaluate historical performance of any model on a given ticker using past data.

---

## 📁 Files Created

- `app/services/backtest.py`: Main logic to simulate prediction vs. actuals and return performance metrics.
- `app/services/data.py`: Fetches and prepares historical data.
- `app/services/models.py`: Loads and executes model predictions (XGBoost, LSTM, Linear).
- `scripts/test_backtest.py`: Manual test script with 3 cases (valid, fake ticker, bad model).

---

## 🧠 Functional Summary

The `backtest()` function accepts:
```python
backtest(ticker: str, model: str, days: int = 30)
